ui/packages/catalog-ui-search/src/main/webapp/react-component/location/location.js

```diff
@@ -103,17 +145,63 @@ const ddValidators = {
   east: value => value <= 180 && value >= -180,
 }
 
+let isDms = false
+const dmsValidators = {
+  dmsLat: value => validateInput(value, 'dd°mm\'ss.s"'),
+  dmsLon: value => validateInput(value, 'ddd°mm\'ss.s"'),
+  dmsNorth: value => validateInput(value, 'dd°mm\'ss.s"'),
+  dmsSouth: value => validateInput(value, 'dd°mm\'ss.s"'),
+  dmsWest: value => validateInput(value, 'ddd°mm\'ss.s"'),
+  dmsEast: value => validateInput(value, 'ddd°mm\'ss.s"'),
+}
+
+const getNegOrPosLatLon = (key, value) => {
+  if (value < 0) {
+    return -1 * validLatLon[key]
+  } else {
+    return validLatLon[key]
+  }
+}
+
 module.exports = ({ state, setState, options }) => (
   <LocationInput
     {...state}
     onDraw={options.onDraw}
     setState={setState}
     cursor={key => value => {
-      const validateCoords = ddValidators[key]
-      if (typeof validateCoords === 'function' && !validateCoords(value)) {
-        return
+      isDms = false
+      let validateCoords = ddValidators[key]
+      if (validateCoords === undefined) {
+        validateCoords = dmsValidators[key]
+        isDms = true
+      }
+      if (!isDms) {
+        if (typeof validateCoords === 'function' && !validateCoords(value)) {
+          errors = true
+          inValidInput = value
+          inValidKey = readableNames[key]
+          defaultCoord = getNegOrPosLatLon(key, value)
+          value = defaultCoord
+          setState(key, value)
+          return
+        }
+        setState(key, value, (errors = false))
```

Comment:
❓ What is this syntax? I thought normally you passed in an object into `setState`

Reply:
It's not the actual React `setState` it's being passed in as a prop from somewhere else.